### PR TITLE
DM explanation in documentation and dispersion slope as a derived quantity

### PR DESCRIPTION
**Pull request by abhisrkckl**
See issue #742 for a detailed discussion.

### CODE REVIEW

File: docs/explanation.rst [hunk 1]
```diff
@@ -183,6 +183,28 @@ not (this is a particular concern when working with X-ray or gamma-ray data).
...
 
+Dispersion Measure
+------------------
+Radio waves emitted by the pulsar experience dispersion as they travel through the ionized 
+interstellar medium (ISM). The time delay due to the interstellar dispersion is given by
+:math:`\frac{K\times DM}{\nu^2}`, where :math:`\nu` is the frequency of the radio signal.
+The dominant source of this dispersion is the presence of free electrons in the ISM, and 
+to a first approximation, the DM can be interpreted as the electron column density along the 
+line of sight to the pulsar. :math:`K` is known as the DM constant, and should be equal to 
+:math:`\frac{e^2}{8 \pi ^2 c \epsilon _0 m_e} \approx 1.3445365918(9)E-07\; m^2/s` based
```
Comment (dlakaplan) [inline docs/explanation.rst:194]: Could you do `E-07` -< `\times 10^{-7}`?  The former doesn't format well.  And this is done below also.

**Comment by dlakaplan** [comment]
Is it worth adding a method to the various dispersion models to convert the DM to that implied by a different value of K?  Or a function in `pint.utils`?  Might be too confusing, or might get into some of the messy definition issues.

File: docs/explanation.rst (hunk 1 above)
Comment (abhisrkckl) [reply docs/explanation.rst]: Done.

**Comment by abhisrkckl** [comment]
I was thinking of adding a new derived quantity dispersion_slope = DMconst * DM. We can also have a utility function convert_dispersion_measure(dm, dmconst_new) which changes the DM for a different value of DMconst.

**Comment by dlakaplan** [comment]
I think that would be a good idea.

**Comment by dlakaplan** [comment]
Would it be possible to give a quick example of using the new methods in the docs rather than just a mention?

Separately, how about making a `dispersion_slope` method within the `Dispersion` model and separately within the `TimingModel`?  So that `m.total_dm(t)` could also be accompanied by `m.total_dispersion_slope(t)`?  I think this would be pretty simple.

**Comment by dlakaplan** [comment]
That may not be needed - I don't know if people will more easily find a method within the `TimingModel` or a separate function

**Comment by abhisrkckl** [comment]
A function within TimingModel will be easier to find. `m.total_dispersion_slope(t)`, similar to `m.total_dm(t)` should be self-explanatory and will show up in code completion better than a similar function in `derived_quantities`.
I think it will be a good idea to implement it, especially since it's pretty simple.

**Comment by dlakaplan** [comment]
Separately I was wondering if some of those methods should also be able to operate on `astropy.Time objects`, and not just TOAs. This might make for a simpler interface for some use cases, and the difference between a TOA and a more generic Time is almost never relevant for DM calculations.

This might be better left for a separate discussion/PR

**Comment by abhisrkckl** [comment]
> Separately I was wondering if some of those methods should also be able to operate on astropy.Time objects, and not just TOAs. This might make for a simpler interface for some use cases, and the difference between a TOA and a more generic Time is almost never relevant for DM calculations.

This won't work if there are DM jumps since they need TOA metadata to work.

**Comment by dlakaplan** [comment]
I think you need to update the docs to include your new methods

**Comment by abhisrkckl** [comment]
I have updated the docs.

**Comment by dlakaplan** [comment]
Looks good.  I'll merge if you are done.

**Comment by abhisrkckl** [comment]
I am done.

**Pull request merged**